tests/util/test_networks.py

```diff
@@ -192,3 +187,41 @@ def test_input_validation_on_ema_norm():
         networks.EMANorm(128, decay=-0.1)
 
     networks.EMANorm(128, decay=0.05)
+
+
+def test_ema_norm_batch_correctness():
+    norm_for_incremental = networks.EMANorm(256, 0.5)
+    norm_for_batch = networks.EMANorm(256, 0.5)
+    norm_for_incremental.train(), norm_for_batch.train()
+
+    with th.random.fork_rng():
+        th.random.manual_seed(42)
+        random_tensor = th.randn(64, 256)
+        for _ in range(100):
+            # calculating EMA of a moving distribution
+            random_tensor += 1
+            norm_for_batch(random_tensor)
+
+        # seeding torch's random module again so that the random permutation
+        # is exactly the same for both incremental and batch method
+        th.random.manual_seed(42)
```

Comment:
Almost identical to code block above -- could factor into helper method and just pass a different method to evaluate (`norm_for_batch` vs `norm_for_incremental.update_stats_incremental`).